infor-design/enterprise
# 920 - Improve Multiselect filtered selection process

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR makes a few changes to the way the Multiselect component handles selecting items from a filtered list:

- Selecting an item no longer resets the value of the Search Input
- Selecting an item no longer rebuilds the Multiselect's list

These changes were made to improve the usability of the Multiselect, keeping the state of the Search input in-tact between selections, and to prevent too much context switching.  

Although we'd originally talked about more drastic changes to the Multiselect's list, and potentially a different way to display selected items, I opted to keep the original design more in-tact while changing a bit of the usability.  In the new configuration, the list takes the role of showing what's currently selected instead of the search input.

This PR also fixed a minor unrelated bug where an AJAX request that populates a Dropdown/Multiselect containing items with no `label` attributes could result in every list item becoming `undefined`.

**Related github/jira issue (required)**:
Closes #920 

**Steps necessary to review your pull request (required)**:

_To tests the multiselect improvements:_
- Pull this branch, run app
- Open http://localhost:4000/components/multiselect/example-index.html
- Open the first Multiselect at the top
- Type "Co" to filter the list
- Select "Colorado".  The item should become selected, but the list and search input should not reset.
- Backspace until the list is completely empty.  The list should reset and "Colorado" should be selected along with the initially selected "Alaska"
- Press the Escape key.  When the list closes, the in-page pseudo-element should read "Alaska, Colorado".

_To test the `undefined` problem:_
- Open http://localhost:4000/components/multiselect/test-selected-rendering-styles.html
- Open any of the Multiselect components on the page.  None should show `undefined` for all the selectable options.

**Included in this Pull Request**:

----

Also requesting we get a look from @picitelli and @fitzorama, who had comments on the original ticket.
